compiler/runtime.go

```diff
@@ -644,15 +644,91 @@ func (e *unaryEvaluator) Type() semantic.MonoType {
 	return e.t
 }
 
+func doUnary(mt semantic.MonoType, op ast.OperatorKind, v values.Value) (values.Value, error) {
```

Comment:
Lightly modified - this was formerly a closure defined inside the unary evaluator. Extracted so it could be shared with the vector version for the vec repeat case.